src/inc/engine/game_component.h

```diff
@@ -4,15 +4,22 @@
 #include "render_component.h"
 #include <vector>
 #include <iostream>
+#include "input_event.h"
 
 class GameComponent
 {
 public:
 
   virtual void update() = 0;
 
+  virtual void update( InputEvent& event ) = 0;
+
+  virtual bool accepting_input() = 0;
+
   virtual std::vector<std::unique_ptr<RenderComponent>>& get_render_components() = 0;
 
+  virtual int get_frames_per_input() = 0;
```

Comment:
Should be removed. The frames per update/input was a crude way to implement updating on certain frames